CMakeLists.txt

```diff
@@ -0,0 +1,934 @@
+	builtin/upload-pack.c builtin/var.c builtin/verify-commit.c builtin/verify-pack.c 
+	builtin/verify-tag.c builtin/worktree.c builtin/write-tree.c)
+
+add_executable(git git.c ${git_SOURCES})
+target_link_libraries(git common-main )
+
+
+add_executable(git-credential-store credential-store.c)
+target_link_libraries(git-credential-store common-main)
+
+add_executable(git-daemon daemon.c)
+target_link_libraries(git-daemon common-main)
+
+add_executable(git-fast-import fast-import.c)
+target_link_libraries(git-fast-import common-main)
+
+add_executable(git-http-backend http-backend.c)
+target_link_libraries(git-http-backend common-main)
+
+add_executable(git-sh-i18n--envsubst sh-i18n--envsubst.c)
+target_link_libraries(git-sh-i18n--envsubst common-main)
+
+add_executable(git-shell shell.c)
+target_link_libraries(git-shell common-main)
+
+if(CURL_FOUND)
+	add_library(http_obj OBJECT http.c)
+
+	add_executable(git-imap-send imap-send.c)
+	target_link_libraries(git-imap-send http_obj common-main ${CURL_LIBRARIES})
+
+	add_executable(git-http-fetch http-walker.c http-fetch.c)
+	target_link_libraries(git-http-fetch http_obj common-main ${CURL_LIBRARIES})
+
+	add_executable(git-remote-http http-walker.c remote-curl.c)
+	target_link_libraries(git-remote-http http_obj common-main ${CURL_LIBRARIES} )
+
+	if(EXPAT_FOUND)
+		add_executable(git-http-push http-push.c)
+		target_link_libraries(git-http-push http_obj common-main ${CURL_LIBRARIES} ${EXPAT_LIBRARIES})
+	endif()
+endif()
+
+add_executable(git-remote-testsvn remote-testsvn.c)
+target_link_libraries(git-remote-testsvn common-main vcs-svn)
+
+if(NOT WIN32)
+	add_executable(git-credential-cache credential-cache.c)
+	target_link_libraries(git-credential-cache common-main)
+
+	add_executable(git-credential-cache--daemon credential-cache--daemon.c)
+	target_link_libraries(git-credential-cache--daemon common-main)
+endif()
+
+set(git_builtin_extra
+cherry cherry-pick format-patch fsck-objects
+init merge-subtree restore show
+stage status switch whatchanged)
+
+#Creating symlinks
+
+foreach(s ${git_SOURCES} ${git_builtin_extra})
+	string(REPLACE "builtin/" "" s ${s})
+	string(REPLACE ".c" "" s ${s})
+	add_custom_command(OUTPUT git-${s}${EXE}
+						COMMAND ${CMAKE_COMMAND} -E create_symlink ${CMAKE_BINARY_DIR}/git${EXE} ${CMAKE_BINARY_DIR}/git-${s}${EXE}
+						DEPENDS git)
+	list(APPEND git_symlinks ${CMAKE_BINARY_DIR}/git-${s}${EXE})
+endforeach()
+
+if(CURL_FOUND)
+	set(remote_exes 
+		git-remote-https git-remote-ftp git-remote-ftps)
+	foreach(s ${remote_exes})
+		add_custom_command(OUTPUT ${s}${EXE}
+						COMMAND ${CMAKE_COMMAND} -E create_symlink ${CMAKE_BINARY_DIR}/git-remote-http${EXE} ${CMAKE_BINARY_DIR}/${s}${EXE}
+						DEPENDS git-remote-http)
+		list(APPEND git_http_symlinks ${CMAKE_BINARY_DIR}/${s}${EXE})
+	endforeach()
+endif()
+
+add_custom_target(git-symlinks ALL DEPENDS ${git_symlinks} ${git_http_symlinks})
+
+#creating required scripts
+if(WIN32)
+	set(LOCALEDIR /mingw64/share/locale)
```

Comment:
Maybe we can use the `FALLBACK_RUNTIME_PREFIX` here, and drop the `if(WIN32)`?

Reply:
sure